### PR TITLE
Upgrade Felix, OSGi and JRE.

### PR DESCRIPTION
**Pull request by pavelhoral**
## Dependency changes

* ✔ JRE version was upgraded to 1.8.
* ✔ Felix framework upgraded to 6.0.x.
* ❌ Left a lot of dependencies at their old versions (e.g. Gogo Shell). We should continue with the upgrade after this gets merged.

## Implementation changes (OSGi specific)

* ✔ All Felix SCR Annotations were replaced with standard OSGi DS annotations.
* ✔ Previous implementation was using `factory=true` with `suppressMetatypeWarning` properties to generate empty OCD. Such construct is not possible with DS and I am not sure what purpose it had. Those properties were removed.
* ✔ Previous Felix SCR Maven plugin was automatically generating `service.pid` based on component name. This property is used on a lot of places in service lookup / filters. This approach is not possible with DS (see [FELIX-5304](https://issues.apache.org/jira/browse/FELIX-5304)). Service PID properties were removed where possible, however additional cleanup is still needed.

## Bundle changes (metadata)

* ❌ Link to Creative Commons license disappeared from `Bundle-License` manifest attribute. This was not an intentional change, but it probably doesn't matter as CC-BY-NC was for docbook documentation.
* ✔ Newer build produces more exhaustive `Import-Packages` configuration. Seems to be in the direction of best-practices and should be ok.

## Known (unrelated) issues

* Current `master` branch is broken due to wrensecurity/wrensec-commons#22 (not caused by the upgrade).

### CODE REVIEW

**Comment by pavelhoral** [comment]
Seems like Semaphore has some issues with the build (will need to investigate a bit). I know I will have to deal with PGP artifact signatures, but the project builds and runs fine on my computer.

File: openidm-zip/src/main/resources/conf/config.properties [hunk 1]
```diff
@@ -92,8 +92,8 @@ org.ops4j.pax.web.config.file=&{launcher.project.location}/conf/jetty.xml
 org.osgi.service.http.enabled=true
 org.osgi.service.http.secure.enabled=true
 
-# Disable the file configuration uncomment these options and enable the bootstrap in the system.properties.
-#felix.fileinstall.disableConfigSave=false
+# Disable file configuration update
+felix.fileinstall.enableConfigSave=false
```
[the remark this answers is not in the document]
Comment (pavelhoral) [reply openidm-zip/src/main/resources/conf/config.properties:96]: Taking this back... we actually need to disable fileinstall's default persistence. Not sure if this disables JSONCongiInstaller's file update. Hopefully not.

**Comment by pavelhoral** [comment]
Every issue has been hopefully resolved (rebased & squashed). PR is ready to be merged.

**Review by karelmaxa — APPROVED** [review]
LGTM!

**Pull request merged**